botorch/utils/objective.py

```diff
@@ -83,9 +83,11 @@ def apply_constraints_nonnegative_soft(
     Returns:
         A `n_samples x b x q (x m')`-dim tensor of feasibility-weighted objectives.
     """
+    if type(eta) != Tensor:
+        eta = torch.as_tensor([eta for _ in range(len(constraints))])
     obj = obj.clamp_min(0)  # Enforce non-negativity with constraints
-    for constraint in constraints:
-        constraint_eval = soft_eval_constraint(constraint(samples), eta=eta)
+    for constraint, e in zip(constraints, eta):
+        constraint_eval = soft_eval_constraint(constraint(samples), eta=e)
```

Comment:
What happens here if `eta` is a single element tensor? I think this would ignore all the constraints after the first one. We should either add a check that they're the same length and error out, or extend the if statement above to catch single element tensors.

Reply:
I forgot to update it, will change it ;)

Reply:
I think the best is to just error out.

Reply:
I fixed it by catching it in `apply_constraints_nonnegative_soft`.